src/EventSubscriber/RequestSubscriber.php

```diff
@@ -271,12 +283,63 @@ protected function constructQuerySpan($connection, $driver, array $query) {
    *   An array of options to pass to the PHP Agent. Ie. tags.
    */
   protected function prepareAgentOptions() {
+    $tags = [];
+
     $route_options = $this->routeMatch->getRouteObject()->getOptions();
     if (isset($route_options['_admin_route'])) {
-      return ['tags' => ['is_admin_route' => TRUE]];
+      $tags['is_admin_route'] = TRUE;
+    }
+
+    // Add tags based on the path patterns configured.
+    $tags = $tags + $this->preparePathPatternTags();
+
+    return $tags;
+  }
+
+  /**
+   * Provide tags based on the path patterns configured.
+   *
+   * @return array
+   *   An array of tags to pass to the PHP Agent.
+   */
+  protected function preparePathPatternTags() {
+    $tags = [];
+
+    // Fetch the current path from route object.
+    $route_object = $this->routeMatch->getRouteObject();
+    $path = $route_object->getPath();
+
+    // Fetch the configured path patterns.
+    $tag_config = $this->apiService->getTagConfig();
+    $path_pattern = $tag_config['path_patterns'];
+
+    if (empty($path_pattern)) {
+      return;
+    }
```

Comment:
**CHANGE REQUEST**

If the configuration might not contain the `path_pattern` key at all, then you'd get a PHP warning here and you need to to check if the key exists before using it:

```
$tag_config = $this->apiService->getTagConfig();
if (empty($tag_config['path_pattern'])) {
  return;
}
$path_pattern = $tag_config['path_patterns'];
```

If the key will always exist, you don't need to add the overhead of `empty`, or even `isset`:

```
$path_pattern = $this->apiService->getTagConfig()['path_patterns'];
if (!$path_pattern) {
  return;
}
```